front/lib/resources/storage/models/group_memberships.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  CreationOptional,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+} from "sequelize";
+import { DataTypes, Model, Op } from "sequelize";
+
+import { User } from "@app/lib/models/user";
+import { Workspace } from "@app/lib/models/workspace";
+import { frontSequelize } from "@app/lib/resources/storage";
+import { GroupModel } from "@app/lib/resources/storage/models/groups";
+
+export class GroupMembershipModel extends Model<
+  InferAttributes<GroupMembershipModel>,
+  InferCreationAttributes<GroupMembershipModel>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare startAt: Date;
+  declare endAt: Date | null;
+
+  declare groupId: ForeignKey<GroupModel["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+
+  static async checkOverlap(
+    userId: number,
+    groupId: number,
+    startAt: Date,
+    endAt: Date | null,
+    excludeId?: number
+  ) {
+    const where: any = {
+      userId,
+      groupId,
+      [Op.or]: [{ endAt: null }, { endAt: { [Op.gt]: startAt } }],
+      startAt: { [Op.lt]: endAt || new Date(8640000000000000) }, // Max date if endAt is null
+    };
+
+    if (excludeId) {
+      where.id = { [Op.ne]: excludeId };
+    }
+
+    const overlapping = await this.findOne({ where });
+    return !!overlapping;
+  }
+}
+GroupMembershipModel.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    startAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    endAt: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "group_memberships",
+    sequelize: frontSequelize,
+    indexes: [{ fields: ["userId", "groupId"] }],
```

Comment:
Unclear yet exactly how we will most often query those but we can update the indexes later if needed